python_modules/dagster/dagster/_core/ext/context.py

```diff
@@ -38,12 +40,17 @@ def handle_message(self, message: ExtMessage) -> None:
         elif message["method"] == "log":
             self._handle_log(**message["params"])  # type: ignore
 
-    def _handle_report_asset_metadata(self, asset_key: str, label: str, value: Any) -> None:
+    def _handle_report_asset_metadata(
+        self, asset_key: str, label: str, value: Any, type: ExtMetadataType  # noqa: A002
+    ) -> None:
         check.str_param(asset_key, "asset_key")
         check.str_param(label, "label")
+        check.opt_literal_param(type, "type", get_args(ExtMetadataType))
```

Comment:
does `get_args` cache?

Reply:
I don't know but it's not slow, compared to directly looking up a list on an object (10000 times each):

```
Time taken for get_args_from_literal: 0.002764 seconds
Time taken for get_array_from_object: 0.001988 seconds
```